### PR TITLE
Midround bomberman

### PR DESCRIPTION
**Pull request by ShiftyRail**
[content] [gameplay] [wip]

## General idea

There's often not a lot to do when you're a ghost, but still want a way to be kept in the round somehow, or if you're waiting for the round to end.
This PR tries to make this experience less dull, by allowing ghosts to create their own bomberman arenas midround, and by changing the way bombermen are generated using transmorgification.

With a transmog, the original body & mind is kept, so if you're suddenly being cloned in the middle of a bomberman match, you can dip in back in the round.

## To be tested

Everything. Dream Deamon isn't playing nice with me and testing this with multiple accounts is a pain.
I'll ask some anons to hop on the test server maybe tomorrow to see if there are any major issues.

I also want to check if the "You're being cloned!" message still applies when transmorgified, and if it doesn't, make it so.

:cl:
- rscadd: Ghosts can now freely create midround bomberman arenas.
- tweak: Joining  a bomberman arena will no longer prevent you from being cloned or brought back as your original character.

### CODE REVIEW

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
I'll take midround thunderdome any day of the week before this, thanks

**Comment by SonixApache** [comment]
good gateway to arena datums and more things to do for ghosts, good idea

**Comment by ShiftyRail** [comment]
Why not have both ?

**Comment by Exxion** [comment]
Out of curiosity, what happens if you get revived while actively playing Bomberman? Does the Bomberman just go braindead/catatonic/etc.?

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -1069,3 +1069,19 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
+	set desc = "Create a bomberman arena for other observers and dead players."
+
+	if (ticker && ticker.current_state != GAME_STATE_PLAYING)
+		to_chat(src, "<span class ='notice'>You can't use this verb before the game has started.</span>")
+
+	if (arenas.len)
+		to_chat(src, "<span class ='notice'>There are already bomberman arenas! Use the Find Arenas verb to jump to them.</span>")
+
+	to_chat(src, "<span class='notice'>Pooling other ghosts for a bomberman arena...</span>")
+	new /datum/bomberman_arena(locate(250, 250, 2), pick("15x13 (2 players)","15x15 (4 players)","39x23 (10 players)"), src)
```
Comment (DeityLink) [inline code/modules/mob/dead/observer/observer.dm]: Why not search for a landmark that can then be placed on Z2?

**Comment by ShiftyRail** [comment]
@Exxion The bomberman just gets deleted. His gear falls on the ground, and he's removed from the arena's players. (as far as I'm aware)

**Comment by ShiftyRail** [comment]
Fuck mappers and fuck conflicts

**Comment by ShiftyRail** [comment]
Oh God oh fuck

**Comment by ShiftyRail** [comment]
@MadmanMartian You (or someone else) broke the line endings in Roid
Reeee

**Comment by Kurfursten** [comment]
What is the status on this?

**Comment by ShiftyRail** [comment]
Need to make it so it doesn't break Roid station maybe

**Comment by SonixApache** [comment]
conflicts

**Comment by ShiftyRail** [comment]
Fucking conflicts
I'll just use a hardmapped location, soft-mapping the landmark can come next time

**Review by MadmanMartian — CHANGES_REQUESTED** [review]
use the dungeon loading function

**Comment by ShiftyRail** [comment]
Bomberman arenas aren't dungeons, they're created and destroyed on the fly, which makes sense as they're supposed to be different every round and there's 3 different types of them

I don't understand what you're asking me to do here.

**Comment by ShiftyRail** [comment]
Not feasible at this time, bomberman arenas are not a map element and are dynamically generated. The dungeon loader was not made for this kind of problems.

**Comment by Kurfursten** [comment]
Is this tested?

**Comment by ShiftyRail** [comment]
Yes it is. I even got players to test it on the test server.

**Pull request merged**